### PR TITLE
Test/impairment capital account bug

### PR DESCRIPTION
**Pull request by bengobeil**

### CODE REVIEW

File: contracts/BullaFactoring.sol [hunk 1]
```diff
@@ -956,18 +963,37 @@ contract BullaFactoringV2_2 is IBullaFactoringV2_2, ERC20, ERC4626, Ownable {
...
         uint256 poolOwnedWithheld = _approval.fundedAmountGross - _approval.fundedAmountNet - ApprovalPacking.protocolFee(_approval) - ApprovalPacking.insurancePremium(_approval);
-        paidInvoicesGain += _impairmentNetGain + poolOwnedWithheld;
+        // When accrued fees exceed what insurance covers (impairmentGrossGain),
+        // the overage must come from the withheld target fees (poolOwnedWithheld).
+        uint256 totalFeesOwed = adminFeeOwed + spreadOwed;
+        uint256 lpCredit;
+        if (totalFeesOwed > _impairmentGrossGain) {
+            uint256 feeOverage = totalFeesOwed - _impairmentGrossGain;
+            // Cap the fee charge at what's available (insurance + withheld)
+            uint256 feeFromWithheld = feeOverage > poolOwnedWithheld ? poolOwnedWithheld : feeOverage;
```
Comment (bengobeil) [inline contracts/BullaFactoring.sol]: I think impairment, gross gain plus pool owned withheld should be added together as gross LP credit, and then you execute the logic of total fees owned, blah blah blah. The same. It doesn't make sense to only withdraw fees from one thing and then try to withdraw from the other; it doesn't make sense, so it is better to add both of those numbers together up front.

File: contracts/BullaFactoring.sol [hunk 2]
```diff
@@ -956,18 +963,30 @@ contract BullaFactoringV2_2 is IBullaFactoringV2_2, ERC20, ERC4626, Ownable {
...
+        uint256 totalFeesOwed = adminFeeOwed + spreadOwed;
+        uint256 grossLPCredit = _impairmentGrossGain + poolOwnedWithheld;
+        uint256 feesCharged = totalFeesOwed > grossLPCredit ? grossLPCredit : totalFeesOwed;
+        adminFeeBalance += feesCharged;
+        uint256 lpCredit = grossLPCredit - feesCharged;
+        // Record the net principal loss: funded capital minus partial payments already
+        // received and any recovery from insurance/withheld fees.
+        // paidInvoicesGain is not touched — it tracks only realised interest.
+        uint256 remainingPrincipal = _approval.fundedAmountNet > currentPaidAmount
+            ? _approval.fundedAmountNet - currentPaidAmount
```
Comment (bengobeil) [inline contracts/BullaFactoring.sol]: I don't think this is correct. Funded amount net takes into account the initial paid amount, but we are not correcting for the initial paid amount here, only the current one. Write a test to confirm the bad behaviour and then fix it.

File: contracts/BullaFactoring.sol [hunk 3]
```diff
@@ -956,18 +963,30 @@ contract BullaFactoringV2_2 is IBullaFactoringV2_2, ERC20, ERC4626, Ownable {
...
+            : 0;
+        uint256 _principalLoss = remainingPrincipal > lpCredit
+            ? remainingPrincipal - lpCredit
```
Comment (bengobeil) [inline contracts/BullaFactoring.sol]: Also, it would be easier to read if we use "payments since funding", which is what it should be called. By the way, this current paid amount, we're actually looking for payments since funding should be subtracted from funded amount net, and LP credit should also be subtracted from funded amount net at the same time. Ideally, it's easier to read that the loss is actually the full funded amount net minus the LP credit and minus any payments made since funding.  
  
We also need to write a test for edge cases, like if an invoice has an extreme amount of late fee interest. I'm thinking it's possible for an invoice to be almost fully paid, so it has a lot of interest, so the paid amount is actually bigger than the funded amount net. This would mean that the impairment loss could actually be less than zero, so I want you to look into that as well.

File: test/foundry/TestInsurance.t.sol [hunk 4]
```diff
@@ -862,4 +862,482 @@ contract TestInsurance is CommonSetup {
...
+        // Warp to 200 days so accrued fees exceed impairmentGrossGain
+        // dueBy = +30d, impairmentGracePeriod = 60d, so 200d is well past
+        vm.warp(block.timestamp + 200 days);
+
+        // Preview to verify fees > grossGain
+        (
+            uint256 outstandingBalance,
+            uint256 impairmentGrossGain,
+            uint256 adminFeeOwed,
+            uint256 impairmentNetGain,
```
Comment (bengobeil) [inline test/foundry/TestInsurance.t.sol]: I'm not sure impairment net gain is really a concept anymore. Can you double check that we actually need this?

Comment (GeraldBennyClawBot) [reply test/foundry/TestInsurance.t.sol]: You're right -- impairmentNetGain is no longer a meaningful concept with the new combined-pool approach. The actual LP credit is now calculated as grossLPCredit - feesCharged where grossLPCredit = impairmentGrossGain + poolOwnedWithheld. Since fees are deducted from this combined pool (not just from the insurance payout), impairmentNetGain (which was grossGain - fees) gives an incomplete picture. Removed it from previewImpair() return values, the InvoiceImpaired event, the interface, and all test references. All 642 tests pass. See commit 99f6f91.

Comment (GeraldBennyClawBot) [reply test/foundry/TestInsurance.t.sol]: You're right -- impairmentNetGain is no longer meaningful after the refactor that combines grossGain + poolOwnedWithheld as gross LP credit before deducting fees. Removed it from previewImpair return value, InvoiceImpaired event, interface, and all test references. Pushed in 99f6f91, CI passed.

File: contracts/interfaces/IBullaFactoring.sol [hunk 5]
```diff
@@ -98,7 +98,7 @@ interface IBullaFactoringV2_2 {
     event InsurerChanged(address indexed oldInsurer, address indexed newInsurer);
     event InsuranceParamsChanged(uint16 insuranceFeeBps, uint16 impairmentGrossGainBps, uint16 recoveryProfitRatioBps);
     event InsuranceWithdrawn(address indexed insurer, uint256 amount);
-    event InvoiceImpaired(uint256 indexed invoiceId, uint256 outstandingBalance, uint256 impairmentGrossGain, uint256 impairmentNetGain);
+    event InvoiceImpaired(uint256 indexed invoiceId, uint256 outstandingBalance, uint256 impairmentGrossGain);
```
Comment (bengobeil) [inline contracts/interfaces/IBullaFactoring.sol]: I think we need more information on this event. Like true admin fee, principal lost.

Comment (GeraldBennyClawBot) [reply contracts/interfaces/IBullaFactoring.sol]: Added feesCharged (actual admin+spread deducted from the combined LP credit pool) and principalLoss (net capital loss after insurance, withheld fees, and partial payments) to the event. Pushed in 3df4c19.
